<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>propModel</class>
 <widget class="QWidget" name="propModel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>200</width>
    <height>235</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>200</width>
    <height>235</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>200</width>
    <height>235</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QGroupBox" name="groupBoxModel">
   <property name="geometry">
    <rect>
     <x>3</x>
     <y>0</y>
     <width>194</width>
     <height>231</height>
    </rect>
   </property>
   <property name="title">
    <string>Model transformation</string>
   </property>
   <widget class="QCheckBox" name="checkCentered">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>204</y>
      <width>171</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Move to center of workspace.</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>120</y>
      <width>34</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale x</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>111</x>
      <y>19</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Translation</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinRotX">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>38</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255,0,0)</string>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>18</x>
      <y>20</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotation</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinRotY">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>63</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(0, 128, 0);</string>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinRotZ">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>88</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(0,0,255)</string>
    </property>
    <property name="suffix">
     <string>°</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinTransX">
    <property name="geometry">
     <rect>
      <x>97</x>
      <y>39</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255,0,0)</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinTransY">
    <property name="geometry">
     <rect>
      <x>97</x>
      <y>64</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(0, 128, 0);</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinTransZ">
    <property name="geometry">
     <rect>
      <x>97</x>
      <y>88</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(0,0,255)</string>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="minimum">
     <double>-999999999.000000000000000</double>
    </property>
    <property name="maximum">
     <double>999999999.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinScale">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>120</y>
      <width>80</width>
      <height>22</height>
     </rect>
    </property>
    <property name="decimals">
     <number>6</number>
    </property>
    <property name="maximum">
     <double>10000000000000000000000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QTableWidget" name="originCoords">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>180</y>
      <width>180</width>
      <height>21</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>59</number>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <attribute name="verticalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderMinimumSectionSize">
     <number>20</number>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>20</number>
    </attribute>
    <row>
     <property name="text">
      <string>Nowy wiersz</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string>Nowa kolumna</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Nowa kolumna</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Nowa kolumna</string>
     </property>
    </column>
   </widget>
   <widget class="QComboBox" name="originType">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>67</x>
      <y>157</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>point: [0,0,0]</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>weight center</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>bbox center</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Set pivot at</string>
    </property>
   </widget>
  </widget>
 </widget>
 <tabstops>
  <tabstop>spinRotX</tabstop>
  <tabstop>spinRotY</tabstop>
  <tabstop>spinRotZ</tabstop>
  <tabstop>spinTransX</tabstop>
  <tabstop>spinTransY</tabstop>
  <tabstop>spinTransZ</tabstop>
  <tabstop>spinScale</tabstop>
  <tabstop>checkCentered</tabstop>
 </tabstops>
 <resources>
  <include location="dpVision.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>spinRotX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedRotX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinRotY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedRotY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinRotZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedRotZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinTransX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedTraXYZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinTransY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedTraXYZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinTransZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedTraXYZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkCentered</sender>
   <signal>stateChanged(int)</signal>
   <receiver>propModel</receiver>
   <slot>changedCentered(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinScale</sender>
   <signal>valueChanged(double)</signal>
   <receiver>propModel</receiver>
   <slot>changedScale(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changedRotX(double)</slot>
  <slot>changedRotY(double)</slot>
  <slot>changedRotZ(double)</slot>
  <slot>changedTraXYZ(double)</slot>
  <slot>changedScale(double)</slot>
  <slot>changedCentered(int)</slot>
 </slots>
</ui>
